libs/usb-drive/src/usb_drive.ts

```diff
@@ -161,5 +217,32 @@ export function detectUsbDrive(): UsbDrive {
       await unmountUsbDrive(deviceInfo.mountpoint);
       didEject = true;
     },
+
+    async format(): Promise<void> {
+      const deviceInfo = await getUsbDriveDeviceInfo();
+      if (!deviceInfo) {
+        debug('No USB drive detected, skipping format');
+        return;
+      }
+
+      isFormatting = true;
+      try {
+        if (deviceInfo.mountpoint) {
+          debug('USB drive is mounted, unmounting before formatting');
+          await unmountUsbDrive(deviceInfo.mountpoint);
+          didEject = true;
+        }
+
+        const label = generateVxUsbLabel(deviceInfo.label ?? undefined);
+        await formatUsbDrive(getRootDeviceName(deviceInfo.path), label);
+        debug('USB drive formatted successfully');
+      } catch (error) {
+        debug(`USB drive formatting failed`);
+        throw error;
+      } finally {
+        isFormatting = false;
+        didEject = true; // prevent remount
```

Comment:
We may want use cases in the future where we want to remount immediately, but none exist now. The only use case is the formatting in the VxAdmin frontend.